### PR TITLE
Closes #523: DataverseServiceIT.deleteDataverse test fix

### LINKED ISSUE #523: Fix DataverseServiceIT#deleteDataverse test.

### PR DESCRIPTION
**Pull request by dbojanek**

### CODE REVIEW

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/DataverseServiceIT.java [hunk 1]
```diff
@@ -217,13 +217,13 @@ public void publishDataverse_WithIllegalCommandException() {
     public void deleteDataverse() {
         //given
         loginSessionWithSuperUser();
-        Dataverse unpublishedDataverse = dataverseDao.find(19L);
+        Dataverse unpublishedDataverse = dataverseDao.find(67L);
```
Comment (madryk) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/DataverseServiceIT.java:225]: If dataverse have publicationDate then it is not really unpublished.

Comment (madryk) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/DataverseServiceIT.java:225]: What was the problem with deleting ROOT -> unreleased dataverse?

**Review by madryk — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/DataverseServiceIT.java (hunk 1 above)
Comment (dbojanek) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/DataverseServiceIT.java:225]: In other tests we've enriched previous dataverse (with contacts, files ect.) witch caused "Dataverse not empty" error. It was easier to create empty new dataverse than to change the old one for all tests.

Comment (madryk) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/DataverseServiceIT.java:225]: Ok. So this was normal situation. I think it would be better If we test that situation also.
We could create extra method `deleteDataverse_with_added_datasets()` and check if it will throw error.

File: dataverse-persistence/src/test/resources/dbinit.sql [hunk 2]
```diff
@@ -62,6 +62,13 @@ INSERT INTO dataverserole (id, alias, description, name, permissionbits, owner_i
 
 INSERT INTO dataversefeatureddataverse (id, displayorder, dataverse_id, featureddataverse_id) VALUES (3, 1, 1, 19);
 
+--- ROOT -> simpleDv ---
+
+INSERT INTO dvobject (id, dtype, owner_id, previewimageavailable, protocol, authority, identifier, globalidcreatetime, identifierregistered, storageidentifier, releaseuser_id, publicationdate, creator_id, createdate, modificationtime, permissionmodificationtime, indextime, permissionindextime)
+    VALUES (67, 'Dataverse', 1, false, NULL, NULL, NULL, NULL, false, NULL, 1, NULL, 1, '2019-06-06 08:27:17.531', '2019-06-06 08:27:43.217', '2019-06-06 08:27:17.588', '2019-09-19 14:37:05.558', '2019-09-19 14:37:05.731');
+INSERT INTO dataverse (id, alias, name, affiliation, dataversetype, description, defaultcontributorrole_id, defaulttemplate_id, facetroot, guestbookroot, metadatablockroot, permissionroot, templateroot, themeroot, allowmessagesbanners)
+    VALUES (67, 'simpleDv', 'simpleDv', 'aff', 'ORGANIZATIONS_INSTITUTIONS', NULL, 6, NULL, false, false, true, true, false, true, false);
```
Comment (madryk) [inline dataverse-persistence/src/test/resources/dbinit.sql]: If we name this dataverse `withoutDatasets` then it would be indicator for others that we shouldn't add datasets to this dataverse

**Review by madryk — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/DataverseServiceIT.java (hunk 1 above)
Comment (dbojanek) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/DataverseServiceIT.java:225]: It's more general. Before delete on dataverse we check if it `hasData()`. If it does then the exception is being thrown, else we proceed with deletion. So extra test could check for expected exception for not 'empty' dataverse.

Comment (madryk) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/DataverseServiceIT.java:225]: `hasData()` in context of dataverse is equivalent to situation where dataverse have child dataverse or have some datasets

**Review by madryk — APPROVED** [review]
